libmscore/keysig.cpp

```diff
@@ -140,7 +140,8 @@ void KeySig::layout()
 
 
       // Don't repeat naturals if shown in courtesy
-      if (prevMeasure && prevMeasure->findSegment(SegmentType::KeySigAnnounce, segment()->tick())
+      if (measure() && measure()->header()
+          && prevMeasure && prevMeasure->findSegment(SegmentType::KeySigAnnounce, segment()->tick())
```

Comment:
This is necessary because of the way layout is done in 3.0. Otherwise, key changes to C major / A minor will not be displayed if the key change does not occur at the beginning of a system.

Reply:
Courtesy sigs are created every time there is a key change, and if not needed are disabled only after the measure with the regular keysig is laid out. If there is any other condition which would allow a courtesy sig to remain enabled, it should be included in this _if_ statement.